core-bundle/src/DataContainer/DataContainerOperationsBuilder.php

```diff
@@ -0,0 +1,270 @@
+            'title' => $config['title'],
+            'popup' => $isPopup,
+            'label' => $config['label'],
+            'attributes' => $config['attributes'],
+            'icon' => Image::getHtml($config['icon'], $config['label']),
+            'primary' => (bool) ($config['primary'] ?? false),
+        ];
+    }
+
+    private function generateHref(DataContainerOperation $config, array $record, bool $isPopup): string|null
+    {
+        if (null !== $config->getUrl()) {
+            return $config->getUrl();
+        }
+
+        if (!empty($config['route'])) {
+            $params = array('id' => $record['id']);
+
+            if ($isPopup) {
+                $params['popup'] = '1';
+            }
+
+            return $this->urlGenerator->generate($config['route'], $params);
+        }
+
+        if (isset($config['href'])) {
+            return Backend::addToUrl($config['href'].'&amp;id='.$record['id'].(Input::get('nb') ? '&amp;nc=1' : '').($isPopup ? '&amp;popup=1' : ''));
+        }
+
+        return null;
+    }
+
+    private function isPopup(string $name): bool
+    {
+        return 'show' === $name;
+    }
+
+    /**
+     * Returns true if this was a toggle operation (which is added to $operations).
+     */
+    private function handleToggle(DataContainerOperation $config, string $table, array $record, array $operation, string|null $href): array|null|false
+    {
+        parse_str(StringUtil::decodeEntities($config['href'] ?? $operation['href'] ?? ''), $params);
+
+        if ('toggle' !== ($params['act'] ?? null) || !isset($params['field'])) {
+            return false;
+        }
+
+        // Do not generate the toggle icon if the user does not have access to the field
+        if (
+            (
+                true !== ($GLOBALS['TL_DCA'][$table]['fields'][$params['field']]['toggle'] ?? false)
+                && true !== ($GLOBALS['TL_DCA'][$table]['fields'][$params['field']]['reverseToggle'] ?? false)
+            ) || (
+                DataContainer::isFieldExcluded($table, $params['field'])
+                && !$this->security->isGranted(ContaoCorePermissions::USER_CAN_EDIT_FIELD_OF_TABLE, $table.'::'.$params['field'])
+            )
+        ) {
+            return null;
+        }
+
+        $icon = $config['icon'];
+        $_icon = pathinfo($config['icon'], PATHINFO_FILENAME).'_.'.pathinfo($config['icon'], PATHINFO_EXTENSION);
+
+        if (str_contains($config['icon'], '/')) {
+            $_icon = \dirname($config['icon']).'/'.$_icon;
+        }
+
+        if ('visible.svg' === $icon) {
+            $_icon = 'invisible.svg';
+        } elseif ('featured.svg' === $icon) {
+            $_icon = 'unfeatured.svg';
+        }
+
+        $state = $record[$params['field']] ? 1 : 0;
+
+        if (($config['reverse'] ?? false) || ($GLOBALS['TL_DCA'][$table]['fields'][$params['field']]['reverseToggle'] ?? false)) {
+            $state = $record[$params['field']] ? 0 : 1;
+        }
+
```

Comment:
At this point the new code differs from the old code:

```php
if ($href === null)
{
	$return .= Image::getHtml($config['icon'], $config['label']) . ' ';
}
else
{
	if (isset($config['titleDisabled']))
	{
		$titleDisabled = $config['titleDisabled'];
	}
	else
	{
		$titleDisabled = (\is_array($v['label']) && isset($v['label'][2])) ? \sprintf($v['label'][2], $arrRow['id']) : $config['title'];
	}
```

Where did the `$href === null` part go?

Reply:
to the template. The config array is indentical, but if `href` is empty the template does not render the anchor.